### PR TITLE
fix summation with plates when manually unrolling

### PR DESCRIPTION
**Pull request by iffsid**
Potential fix for manual unrolling of plate

issue: #2361

### CODE REVIEW

**Comment by fehiepsi** [comment]
Thanks, @iffsid! Could you also add the failing example in #2361 to [this](https://github.com/pyro-ppl/pyro/blob/dev/tests/infer/test_inference.py#L747) so we can make sure that future changes do not have regression?

File: tests/infer/test_inference.py [hunk 1]
```diff
@@ -745,3 +745,33 @@ def model():
...
+    svi = SVI(model, guide, adam, loss_fn)
+
+    for step in range(1001):
+        loss = svi.step(data)
+        if step % 20 == 0:
+            logger.info("step {} loss = {:0.4g}".format(step, loss))
+
+    expected_p = torch.tensor(0.5)
+    actual_p = pyro.param("p").detach()
+    assert_close(actual_p, expected_p, atol=0.05)
```
Comment (fehiepsi) [inline tests/infer/test_inference.py:777]: It is surprising to me that without enumeration, SVI works for this pair of `model`/`guide`. Bernoulli is a discrete distribution, so grad can't pass through its samples. What happens if you change the initial parameter `p` to something different from `0.5`, for example `p = pyro.param('p', torch.tensor(0.2))`?

If SVI fails, I would suggest to only run 1 step `svi.step(data)` to verify `RenyiELBO` can run with a sequential plate. Alternatively, I guess you can change `Bernoulli` to some other continuous distribution, for example `dist.Normal(0, 1)` in `model` and `dist.Normal(p, 0.1)` in the guide, to make the inference work.

Comment (iffsid) [reply tests/infer/test_inference.py:777]: Ah, I didn't actually do any of those checks---just put in a dummy check as we didn't care about the answer in this case.
I thought the use of `score_parts` within `RenyiElbo` handles the discrete Bernoulli?

In any case, I'll just run this for one step as we just need to be sure it 'runs'. :)

Comment (fehiepsi) [reply tests/infer/test_inference.py]: You are right! `score_parts` might remedy that issue. Thanks for reminding me about it!

**Pull request merged**